### PR TITLE
README: Update the text and reformat in rst

### PR DESCRIPTION
**Pull request by max-moser**
Previously, the README was formatted in markdown, contradicting the [restructured text](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html) file ending and caused a broken rendering on GitHub.
I've translated the format into actual reST and updated the text a bit.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -29,57 +31,89 @@ Further documentation is available on
...
+Overwrite deposit view template
+-------------------------------
+
+The deposit view has to be updated to include the curation section.
+Most importantly, the curation specific JavaScript has to be included in the JavaScript block:
+``{{ webpack['invenio-curations-deposit.js'] }}``
+
+This can be achieved by providing a custom template, e.g. in your instance's ``templates/`` directory:
+
+Copy the current template from ``invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html`` (available e.g. `here <https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html>`_) into your instance's ``templates/`` directory (the last parts of the path have to match): ``templates/semantic-ui/invenio_app_rdm/records/deposit.html``.
```
Comment (max-moser) [inline README.rst:96]: for the record, one of the reST quirks:

originally i tried to create a link with code formatting:
~~~rest
```invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html`` <https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html>`_
~~~
but it looks like sphinx couldn't handle the triple back-ticks in the beginning and rendered weird stuff.

in markdown, this is simple:
[`invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html`](https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html)

```md
[`invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html`](https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html)
```

File: README.rst [hunk 2]
```diff
@@ -29,57 +31,89 @@ Further documentation is available on
...
-NOTIFICATIONS_BUILDERS = {
-    **NOTIFICATIONS_BUILDERS,
-    # Curation request
-    CurationRequestAcceptNotificationBuilder.type: CurationRequestAcceptNotificationBuilder,
-    CurationRequestSubmitNotificationBuilder.type: CurationRequestSubmitNotificationBuilder,
-}
+Additionally, notification builders have to be configured so that notifications are sent out to the involved users whenever something's happening in the curation review.
 
-NOTIFICATIONS_ENTITY_RESOLVERS = NOTIFICATIONS_ENTITY_RESOLVERS + [ServiceResultResolver(service_id="groups", type_key="group")]
+.. code-block:: python
```
Comment (max-moser) [inline README.rst:48]: note: the empty line between this and the (indented) code block is required; omitting it won't render any code block.

File: README.rst [hunk 3]
```diff
@@ -29,57 +31,89 @@ Further documentation is available on
...
-from invenio_curations.notifications.builders import (
-    CurationRequestAcceptNotificationBuilder,
-    CurationRequestSubmitNotificationBuilder,
-)
+Currently, requests can only be sent to a single `receiver`.
+However, curation reviews are typically performed by a `group` of people rather than one single fixed `user`.
+Thus, the curation requests are sent to a `group` rather than a single `user` in the system so that all users with a certain `role` can receive and act on curation requests.
 
-from invenio_records_resources.references.entity_resolvers import ServiceResultResolver
+To enable resolution of user groups, an `entity resolver` has to be added to the configuration.
```
Comment (max-moser) [inline README.rst:44]: this paragraph is based on educated guesswork, is it even true?

Comment (rekt-hard) [reply README.rst:44]: That sounds correct. Setting the group as receiver allows to access the request. For the record itself, there is an additional access granted to the group on request creation.

**Review by rekt-hard — APPROVED** [review]

**Review by rekt-hard — APPROVED** [review]
Awesome rework - thank you a lot!

**Pull request merged**